docs/overview.rst

```diff
@@ -515,6 +515,36 @@ without extra cost because the result is cached by the system.
 
     assert service.types is types
 
+.. _annotations:
+
+Customizing behavior with annotations
+-------------------------------------
+
+Thrift supports adding annotations to defined types, fields, references to
+primitive types, etc. These may be used to customize behavior of ``thriftrw``.
+
+The following annotations are supported by ``thriftrw``.
+
+``py.primitive_type``
```

Comment:
Don't you think that's misleading. `py.type` would imply this changes the actual type of the values being passed around, but we'll still be passing ints around. This will only change what `to_primitive` does, so if you `SomeStruct(value=MyEnum.Enabled)`, `to_primitive` for that will become `{"value": "Enabled"}`.

But that's a good idea. I could just change the language-level representation instead.